code/game/objects/effects/spawners/random/maintenance.dm

```diff
@@ -23,7 +23,12 @@
 		if(mapload)
 			stack_trace("[src] at [AREACOORD(src)] tried to spawn in a space or openspace tile! Mappers please use /obj/effect/spawner/random/maintenance/no_decals instead! Note: the coords might be for a ruin so keep that in mind.")
 		else
-			new /obj/effect/spawner/random/maintenance/no_decals(loc) // if this happens outside of mapload, just spawn a no_decals version and move on
+			// if this happens outside of mapload, just spawn a no_decals version and move on
+			var/regex/no_decal_regex = regex("maintenance")
+			var/new_path_string = no_decal_regex.Replace("[type]", "maintenance/no_decals")
+			var/no_decals_type = text2path(new_path_string)
+			new no_decals_type(loc)
```

Comment:
why not just use an `istype(type, /obj/effect/spawner/random/maintenance)` check and spawn the no_decals version? What is the purpose of doing a regex check here? Are there several `no_decal` subtypes?

Reply:
Any particular reason to do this path thing instead of just removing decals from own pool and continuing init?